scripts/core/CUDA.rst

```diff
@@ -148,6 +148,36 @@ take the extra global offset argument. Use of the global offset is not
 recommended for non SYCL compiler toolchains. This parameter can be ignored if
 the user does not wish to use the global offset.
 
+Local Memory Arguments
+----------------------
+
+In UR local memory is a region of memory shared by all the work-items in
+a work-group. A kernel function signature can include local memory address
+space pointer arguments, which are set by the user with
+``urKernelSetArgLocal`` with the number of bytes of local memory to allocate
+and make available from the pointer argument.
+
+The CUDA adapter implements local memory arguments to a kernel as a single
+``__shared__`` memory allocation, with each local address space pointer argument
```

Comment:
It's not actually a pointer arg, it ends up as a `u32 offset` arg, which is combined with the ptr within the kernel

Reply:
I mean you actually mention this elsewhere. So maybe this is OK

Reply:
Do you think we could add an overload for `urKernelSetArgLocalExt` that only takes a `Size`, not a `Size, Index`? Then the user could ask for dynamic shared mem without having to have extra offsets in the kernel args.

Reply:
I believe we've an internal ticket about changing this, and combining the ptr+offset on the host side? Is that right, @hdelan?

Reply:
I've rephrased this description to say a `u32` type parameter early in the text.

> Do you think we could add an overload for urKernelSetArgLocalExt that only takes a Size, not a (Size, Index)? Then the user could ask for dynamic shared mem without having to have extra offsets in the kernel args.

I'd probably go for adding an `urKernelSetExecInfo` property for this because at that point, we're not talking about kernel arguments. That might be a viable alternative usage, but I'm not sure how much value it really buys. Defer to @oneapi-src/unified-runtime-cuda-write on if you want us to create an issue somewhere on this topic.

Reply:
Sounds good to me. Nice one